Controllers/ReferenceController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using ReferenceReferralApp.Models;
+
+namespace ReferenceReferralApp.Controllers
+{
+    public class ReferenceController : Controller
+    {
+        private readonly ReferenceContext _context;
+
+        public ReferenceController(ReferenceContext context)
+        {
+            _context = context;
+        }
+
+        // GET: Reference
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.References.ToListAsync());
+        }
+
+        // GET: Reference/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var reference = await _context.References
+                .FirstOrDefaultAsync(m => m.ID == id);
+            if (reference == null)
+            {
+                return NotFound();
+            }
+
+            return View(reference);
+        }
+
+        // GET: Reference/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: Reference/Create
+        // To protect from overposting attacks, enable the specific properties you want to bind to, for 
+        // more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("ID,firstName,lastName,email")] Reference reference)
+        {
+            if (ModelState.IsValid)
+            {
+                _context.Add(reference);
```

Comment:
I would also move your data context to that repository, and throw a using statement around it. The datacontext ends up being a heavy lift. Ensuring it's disposed off properly will help overall performance.